sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -1716,6 +1716,123 @@ public void testIssue1505() {
         );
     }
 
+    @Test
+    public void testArraysOverlap() {
+        this.qs("""
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3], ARRAY [2, 4]);
+                 arrays_overlap
+                ----------------
+                    true
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3], cast(null as integer array));
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2, 3, null], cast(null as integer array));
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(cast(null as integer array), ARRAY [1, 2, 3]);
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(cast(null as integer array), ARRAY [1, 2, 3, null]);
+                 arrays_overlap
+                ----------------
+                    NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [null, 1], ARRAY [2, 1]);
+                 arrays_overlap
+                ----------------
+                 true
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [1, 2], ARRAY [1, null]);
+                 arrays_overlap
+                ----------------
+                 true
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [null, 1], ARRAY [2, 1, null]);
+                 arrays_overlap
+                ----------------
+                 true
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(ARRAY [null, 1], ARRAY [2, 3, null]);
+                 arrays_overlap
+                ----------------
+                 NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(cast(null as integer array), cast(null as integer array));
+                 arrays_overlap
+                ----------------
+                 NULL
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(array[1, 2], array[3]);
+                 arrays_overlap
+                ----------------
+                 false
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(array[3], array[2]);
+                 arrays_overlap
+                ----------------
+                 false
+                (1 row)
+                
+                SELECT ARRAYS_OVERLAP(array [3], array [1, null]);
+                 arrays_overlap
+                ----------------
+                 NULL
+                (1 row)
+                """
+        );
+    }
+
+    @Test @Ignore("https://github.com/feldera/feldera/issues/1475")
+    public void testArraysOverlapDiffTypes() {
```

Comment:
yes, let's make these tests fail in compilation in fact.
document that the arrays must have elements of the same type (excluding nullability).

Reply:
Okay, that simplifies it.

Reply:
can we make this test work by expecting an error?

Reply:
No. Calcite will still evaluate it to `false` for the optimized version.